src/components/Request.js

```diff
@@ -24,23 +16,15 @@ export class RequestMaker extends Component {
 
         // Params that will be used to construct URL.
         params: PropTypes.object,
-
-        // Result handlers.
-        ...Request.handlersPropTypes
-    }
-
-    static defaultProps = {
-        then: () => undefined,
-        catch: () => undefined
     }
 
     static get endpoint() {
         return 'https://kudosapp-staging.herokuapp.com/'
     }
 
     state = {
-        inProgress: false,
-        done: false,
+        loading: false,
+        error: null,
     }
 
     componentWillReceiveProps(nextProps, nextContext) {
```

Comment:
according to this doc this method is considered legacy and we should use `static getDerivedStateFromProps()`

https://reactjs.org/docs/react-component.html

Reply:
Good catch. 🤔 Thanks!